pkg/daemon/daemon.go

```diff
@@ -804,22 +801,39 @@ func (dn *Daemon) checkStateOnFirstRun() error {
 	}
 	// Update our cached copy
 	dn.node = node
-	pendingConfigName, err := dn.getPendingState()
+	pendingState, err := dn.getPendingState()
 	if err != nil {
 		return err
 	}
+	var pendingConfigName, bootID string
+	if pendingState != nil {
+		pendingConfigName = pendingState.Pending
+		bootID = pendingState.BootID
+	}
 	// XXX: drop this
 	// we need this compatibility layer for now
-	if pendingConfigName == "" {
-		pendingConfigName, err = dn.getPendingConfig()
+	if pendingState == nil {
+		legacyPendingState, err := dn.getPendingConfig()
 		if err != nil {
 			return err
 		}
+		pendingConfigName = legacyPendingState.PendingConfig
+		bootID = legacyPendingState.BootID
 	}
+
 	state, err := dn.getStateAndConfigs(pendingConfigName)
 	if err != nil {
 		return err
 	}
+
+	// if we have a pendingConfig but we're into the same bootid, we failed to drain or reboot
+	// and if we still have a pendingConfig it means we've been killed by kube after 600s
```

Comment:
These seem like distinct cases.  If we failed to drain, we should easily be able to detect that.

Failure to *reboot* is where we get killed by kube.

Reply:
uhm according to logs, we _were_ draining, it just took too long so we're in the middle of a drain and can't even get past the Drain function to reboot. That's where we _also_ get sigkill'ed by kubelet

Reply:
a normal failure to drain doesn't incur in the BZ, if we get a Drain error, we just rollback everything and retry, this is about being stuck at drain and getting killed by exceeding the 600s grace term period